### ISSUE TITLE
Login at geocaching.com doesn't work anymore

### ISSUE THREAD

**Issue by CachingFoX** [tomasbedrich/pycaching#162]
After Groundspeak change page structure, login method raise an error. The page header are build up dynamically and  ```Geocaching.get_logged_user()``` doesn't find the user name.

A very quick and dirty hack will be
```
        def get_logged_user(self, login_page=None):
...
        logging.debug("Checking for already logged user.")
        s = login_page.find_all("script")
        for i in s:
            m = re.search(r'"username":."(.*)"', str(i))
            if m:
                return m[1]

        return None
```

**Comment by CachingFoX** [comment]
👍 looks much better than my solution (a late in the night quick'n'dirty workaround)! Thanks!

**Comment by CachingFoX** [comment]
Any updates about this issue?

**Comment by FriedrichFroebel** [comment]
PR #163 should fix this issue, but there still is one failing test. I did not yet have the time to look into this, so if you have an idea about a possible fix of this test, feel free to propose a fix here or in the PR to allow a merge.

**Comment by tomasbedrich** [comment]
Fixed in 4.2.2. Please try and let us know eventually.

**Comment by CachingFoX** [comment]
Thank you very much - log in with 4.2.2 works
